test/functional/tests/lazy_writes/test_flush_huge_dirty_data.py

```diff
@@ -0,0 +1,169 @@
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_flush_over_400_gibibytes_with_fs(cache_mode, fs):
+    """
+        title: Test of the ability to flush huge amount of dirty data with filesystem.
+        description: |
+          Flush cache when amount of dirty data in cache with core with filesystem exceeds 400 GiB.
+        pass_criteria:
+          - Flushing completes successfully without any errors.
+    """
+    with TestRun.step("Prepare devices for cache and core."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([file_size * 1.05])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        Udev.disable()
+
+    with TestRun.step(f"Start cache in {cache_mode} mode."):
+        cache = casadm.start_cache(cache_part, cache_mode)
+
+    with TestRun.step(f"Add core with {fs.name} filesystem to cache and mount it."):
+        core_dev.create_filesystem(fs)
+        core = cache.add_core(core_dev)
+        core.mount(mnt_point)
+
+    with TestRun.step("Disable cleaning and sequential cutoff."):
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+    with TestRun.step("Create test file"):
+        test_file_main = File.create_file("/tmp/test_file_main")
+        fio = (
+            Fio().create_command()
+            .io_engine(IoEngine.libaio)
+            .read_write(ReadWrite.write)
+            .block_size(bs)
+            .direct()
+            .io_depth(256)
+            .target(test_file_main.full_path)
+            .size(file_size)
+        )
+        fio.default_run_time = timedelta(hours=4)  # timeout for non-time-based fio
+        fio.run()
+        test_file_main.refresh_item()
+
+    with TestRun.step("Validate test file and read its md5 sum."):
+        if test_file_main.size != file_size:
+            TestRun.fail("Created test file hasn't reached its target size.")
+        test_file_md5sum_main = test_file_main.md5sum()
+
+    with TestRun.step("Write data to exported object."):
+        test_file_copy = test_file_main.copy(mnt_point + "test_file_copy")
+        test_file_copy.refresh_item()
+        sync()
+
+    with TestRun.step("Check if dirty data exceeded 400 GiB."):
+        minimum_4KiB_blocks = int(Size(400, Unit.GiB).get_value(Unit.Blocks4096))
+        if int(cache.get_statistics().usage_stats.dirty) < minimum_4KiB_blocks:
+            TestRun.fail("There are not enough dirty data in cache!")
+
+    with TestRun.step("Unmount core and stop cache with flush."):
+        core.unmount()
+        try:
+            # this operation could take few hours, depending on core disk
+            output = TestRun.executor.run(stop_cmd(str(cache.cache_id)), timedelta(hours=12))
+            if output.exit_code != 0:
+                raise CmdException("Failed to stop cache.", output)
```

Comment:
I'd prefer TestRun.fail here, because exception can cause stopping the whole tests execution. And this is actually fail when casadm stop command returns non-zero status here